test/functional/rpc_net.py

```diff
@@ -336,7 +336,7 @@ def test_addpeeraddress(self):
 
         self.log.debug("Test that adding an already-present tried address to the new and tried tables fails")
         for value in [True, False]:
-            assert_equal(node.addpeeraddress(address="1.2.3.4", tried=value, port=8333), {"success": False})
+            assert_equal(node.addpeeraddress(address="1.2.3.4", tried=value, port=8333), {"success": False, "error": "failed-adding-to-new"})
```

Comment:
This is a non-exhaustive case now. There are two potential error outcomes that can be triggered on demand, from which only one is being tested:

- a) The address is already in `new` and we are trying to add it again (to any table)
- b) The address is already in `tried` and we are trying to add it again (to any table)

In this case, only `b)` is being tested: we do so both with and without the "try to add to tried" flag, but the reality is that the reason why the function is failing is the entry is already present in `tried`, even when we try to add it to `new`. Given how the addrman works, we cannot really try to add things to `tried` multiple times on demand, given it would need to be added to `new` first and that would fail.

So I think the only way to properly test this would be to add two different addresses twice. e.g:

```python
# Try that the same address cannot be added twice to `new`
assert_equal(node.addpeeraddress(address="1.2.3.5", tried=False, port=8333), {"success": True})
assert_equal(node.addpeeraddress(address="1.2.3.5", tried=False, port=8333), {"success": False, "error": "failed-adding-to-new"})
# Try that the same address cannot be added twice to `tried`
assert_equal(node.addpeeraddress(address="1.2.3.6", tried=True, port=8333), {"success": True})
assert_equal(node.addpeeraddress(address="1.2.3.6", tried=True, port=8333), {"success": False, "error": "failed-adding-to-new"})
```

Reply:
Done!